kubespawner/reflector.py

```diff
@@ -208,7 +208,7 @@ def __init__(self, *args, **kwargs):
 
         self.watch_task = None
 
-    async def _list_and_update(self):
+    async def _list_and_update(self, resource_version=""):
```

Comment:
```suggestion
    async def _list_and_update(self, resource_version=None):
```

Reading through https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions, I see that there's a difference between this being unset, and being set to the string "0" but not for the empty string "". 

So I think we should either use `None` here to indicate that it will be *unset* if not specified, or "0" to get the behavior for that. As "" doesn't seem to be defined, I don't think we should default to that.

I'm not super set on `None` or `"0"`, we can go with whatever you think works better here :)

Reply:
it is not really clear what unset means, in the code resource version seems to always just be a string and it is compared against an empty string which I interpret as meaning unset.

then another issue is what the python api library does with None vs a string but I assume it is likely equivalent so using None may be clearer

Reply:
In the page you linked to, I see a column in the table for `resourceVersion not set`. I am slightly confused about what you mean by "it is not really clear what unset means". By "in the code", do you mean in this codebase or in kubernetes?

Another alternative is to just require `resource_version` to always be passed in here, and not let that be optional.

Reply:
yes I meant the kubernetes code, the resource version there is a string which either has an opaque value (which happens to be etcd's modification index) or is empty

empty means unset